core/src/main/scala/org/broadinstitute/dsde/rawls/submissions/SubmissionsService.scala

```diff
@@ -537,6 +540,11 @@ class SubmissionsService(
         ps.failureMode,
         ps.header
       )
+      _ <- getSetToDelete(submissionRequest)
+        .map { setToDelete =>
+          entityServiceConstructor(ctx).deleteEntities(workspaceName, Seq(setToDelete), None, None)
+        }
+        .getOrElse(Future.successful(()))
```

Comment:
last outstanding issue is to add some error-handling here (or push back and say you don't think we should). Also of note is that `deleteEntities` is only considered successful if it returns an empty set; if it returns values, the deletion failed due to something referencing the entity. See https://github.com/broadinstitute/rawls/blob/5d2c739d3d78a8113cb1b6fbce894f08659c5241/core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/EntityApiService.scala#L183-L187

Reply:
Well if the deletion fails, I don't think it should fail the submission.  We could report on a failure but I'm not sure where that message would go from this point in the code.

Reply:
What failures are you checking here? A `DeleteEntitiesConflictException` will be caught and swallowed so the submission will continue on; but if I am reading the code correctly any other exception (say, a transaction lock-wait timeout) will be thrown and cause the submission to fail.

Reply:
Good point!  Added a recover